cirq/google/calibration/phased_fsim.py

```diff
@@ -523,6 +640,111 @@ def _json_dict_(self) -> Dict[str, Any]:
         }
 
 
+def _get_labeled_int(key: str, s: str):
+    ma = re.match(rf'{key}_(\d+)$', s)
+    if ma is None:
+        raise ValueError(f"Could not parse {key} value for {s}")
+    return int(ma.group(1))
+
+
+def _parse_xeb_fidelities_df(metrics: 'cirq.google.Calibration', super_name: str) -> pd.DataFrame:
+    """Parse a fidelities DataFrame from Metric protos.
+
+    Args:
+        metrics: The metrics from a CalibrationResult
+        super_name: The metric name prefix. We will extract information for metrics named like
+            "{super_name}_depth_{depth}", so you can have multiple independent DataFrames in
+            one CalibrationResult.
+    """
+    records: List[Dict[str, Union[int, float, Tuple['cirq.Qid', 'cirq.Qid']]]] = []
+    for metric_name in metrics.keys():
+        ma = re.match(fr'{super_name}_depth_(\d+)$', metric_name)
+        if ma is None:
+            continue
+
+        for (layer_str, pair_str, qa, qb), (value,) in metrics[metric_name].items():
+            records.append(
+                {
+                    'cycle_depth': int(ma.group(1)),
+                    'layer_i': _get_labeled_int('layer', cast(str, layer_str)),
+                    'pair_i': _get_labeled_int('pair', cast(str, pair_str)),
+                    'fidelity': cast(float, value),
```

Comment:
cast here might break. What I observed (at least the other direction) is that interger-like float values are converted to int somewhere in the pipeline: float(value) will be safer.

Reply:
done